src/include/common/types/types.h

```diff
@@ -239,7 +240,7 @@ class StructTypeInfo : public ExtraTypeInfo {
     void serializeInternal(Serializer& serializer) const override;
 
 private:
-    std::vector<std::unique_ptr<StructField>> fields;
+    std::vector<StructField> fields;
```

Comment:
There still is no possibility of an unintentional copy. StructField has no copy constructor (deleted implicitly since it contains a `unique_ptr`: [E.g.](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGEgMykrgAyeAyYAHI%2BAEaYxBLSAA6oCoRODB7evgGkyamOAqHhUSyx8VK2mPYFDEIETMQEmT5%2BXIF2mA7pdQ0ERZExcQm29Y3N2W0jvf0lZQkAlLaoXsTI7Bzm/mHI3lgA1Cb%2BbgoExGHAh9gmGgCCm9u7mAdHbCwkAJ6X13dmWww7Xn2hzcADdOkRiF9bt9vjsmAoFHs6sQvA4AGJ4KroA4Adist0SXmitDwyBA3z2lKRp1RBAxWIgJ3QIBAJzORj2zDYpD2TJZXgYeAAjl5MAB9RKnYFhAiXPYEd6JTDzPYgTmsTCMgjMkBvMEQLnKxbyxWavm61D6hVK%2BYqkx4%2B0AEW%2BiTOIKYBHYFKp5rZ53VbEO%2BJuVN52v5gpF4slkKOMrl1swQZhOOd/mD3xlexYTDCEDtDtuofNYIcJGByNp9No6Dl/FQyaLVPrADpMCxEgY1mLokxkABrCDmMz14c8805/vigXC0XSwSXCBcW2NkPN1CoFuEhQIHt9wejgCsFi4JkPjvmq6dHEWtE4h94fg4WlIqE4bms1l5y1WT02PFIAhNBvUgICQNAsFwQgSHISgaHoQ1OAA/hBBEMR2AqFD5CUNRgNIXQzH0QxjE/SxKmqdIXAYdxPBaEBCJCMIBlKIYuAADlyFI0gEcY/EIvJuIYGZBnKDiOi6AQejGWjskI8SaikvomNmVixNGJoZL4qZGmEljRMWBQfzWTh7n%2BR5njcJw2UwVgoR%2BP4ASBI4/SMOzTMcv8XnbD43N%2BB5AU8yyUlOGyWDc24s3rCAs3dbxlVxYNQ2ITACBWBg9li0VniufxsD2fwr1Tb4Tg9Uk9jQBgTkwVRXT2LMADUbmCOQ8sOR09jYq8IsEPZe2IaKesy%2BL7USqlktS4h0qGg4zAANj2CBGuavKAFo9mXQrnWhbqCGzXMGHzBLvUpesFq4X5L3TY7eoaM6Ls2kC7w4B9SCfF83w4D9LC/QyVjWGb/DMXggOfG9Fn7EB/H8FsodhuH4dm/ROEkR88I%2B3gFBADRAOAxY4FgcDUA7Og4lgiA0GJ%2Bh4mANouD4OhPWITGIGiPDojCBp3iQ3h2eYYh3gAeWibRwW50gKbYQQBYYWgudB0gsGiLxgDcMRaEx7heCwHMjHEeX8GSrowQ1l9qs6LxPTFmUqjwklomITmPCwPDTjwFhucWKgDGABR6sxAB3AWlSfZDZDQ8RMNkRQVHUeXdDpgwjBQUibDtzHIEWVBJXSDXeEtOIziwdP83I8FKIgVxeL0RjihEvQBJqKu6Yb9JdLmOn5O6dSm9LiTanUtvWO0jSslaYfB/KAyjIw2971R%2BWPr2VQ2NmlbZskPZgGQZB1uhrgFqgiEAeXYHccWBAbKweIS4hqGYfhh%2BocRp6UdetHOAxrGcdBvGYEQEAzbIAtjBCgEAGg%2B2UIYKoQgECoH9iHXgFNEh0FKgISB4RaAwLgW9RBRNkFUxQInGmOINDwiUI0cWeCSbEAiBqD%2BlDKZxAFhbLB8C8KAJuMQH29DAF1HwE%2BXgWFw4YWkFhaOuE456CIcnb6ZE07wEztnAQudXxgmIIXTAxdFgokEG7TAAAVDc6sFFLD%2BhhEY/D0HQNgWwzWpB/YO0SB7JGz157vU4NgVQ5sj7L1XuvcqRD1o4hbBoFs%2B8wEIjiDUBQKoIBfSsGRPYh8SDH3mKfH%2B4NIbQ0fo/Fxr8cGvnoZjbGIMtC2hcUDN%2BC96GlLBqQNRqRnCSCAA%3D)). Given that its purpose is just for creating struct types, I don't think it really needs one either, so we can just leave it as a move-only type.